lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -821,6 +790,16 @@ GenericSignatureRequest::evaluate(Evaluator &evaluator,
       parentSig = extendedNominal->getGenericSignatureOfContext();
       genericParams = nullptr;
     }
+
+    // The inherited entries influence the generic signature of the extension,
+    // because if it introduces conformance to invertible protocol IP, we do not
+    // we do not infer any requirements that the generic parameters to conform
+    // to invertible protocols. This forces people to write out the conditions.
+    if (ext->hasInvertibleLocalConformance())
+      inferInvertibleReqs = false;
```

Comment:
> I'm re-introducing the hack for re-using the signature.

Is this not just a nicer way to express what that loop is doing? I.e., we are also cancelling inference of conformance requirements to invertible protocols upon finding an invertible protocol in the inheritance clause there.

Reply:
The actual change here is the last commit.

Reply:
Oops, sorry, I completely missed that one!